### DISCUSSION TITLE
how to close surfaces and generate volume body

### DISCUSSION THREAD

**Discussion by onsinsin** [gempy-project/gempy#850] [General]
i try to generate volume body in example [18_creating_depth_maps_from_gempy_models](https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/18_creating_depth_maps_from_gempy_models.html),then export 3d models in gltf format to show in Cesium or Threejs, dont know how todo ,need help.

![final effect looks like this](https://github.com/cgre-aachen/gempy/assets/12404840/d602c44d-3811-4a5f-a0e8-999a845684fc)

**Comment by javoha** [comment]
Hi,
first of all - I converted this to a discussion as it does not seem to be an issue or bug. Gempy doese not create watertight volume bodies by default. You can extract the volumes from the resulting grid, but I am not sure if that is what you need.

**Comment by nlosacco** [comment]
Hi @javoha ,

how can volumes be extracted from the grid? In particular I'm worried about the fault zone, where the surface obtained from get_surfaces is staggered (i.e. somewhat it follows the voxels)

**Reply by javoha** [reply]
Hi again,
so maybe I am not 100% sure what exactly you need but I will try to help. The Co-kriging algorithm used in Gempy will create a scalar field per structural group (e.g per fault or grouped layer stack). Each point in space has a value in this field but we either have a discretizes regular grid to evaluate or use refinement. The isosurfaces displayed in plots are extracted from our scalar fields using a marching cube algorithm. This will not result in watertight volumes and how the fault looks in plots will mainly depend on your chosen resolution/refinement level.
You could extract points from your regular grid that belong to a specific structural unit, but I guess that's not what you want?
